### DISCUSSION TITLE
Customize router-generated errors (e.g. 405)

### DISCUSSION THREAD

**Discussion by gar1t** [tokio-rs/axum#2889] [Q&A]
### Summary

The response body for a 405 error (method not allowed) is empty by default. I'd like the body to show the "canonical reason" instead - e.g. `Method Not Allowed`. How would I go about this?

### axum version

0.7.5

**Comment by mladedav** [comment]
Currently, you'd have to add a fallback to each route as far as I know.

What I think you would want is #2251.
